platforms/java/com/area9innovation/flow/HttpSupport.java

```diff
@@ -213,12 +202,29 @@ public final Object httpCustomRequestWithTimeoutNative(String url, String method
 				responseHeaders.add(kv);
 			}
 
-			onResponse.invoke(responseCode, response.toString(), responseHeaders.toArray());
-        } catch (MalformedURLException e) {
-        	onResponse.invoke(400, "Malformed url " + url + " " + e.getMessage(), new Object[0]);
-        } catch (IOException e) {
-        	onResponse.invoke(500, "IO exception " + url + " " + e.getMessage(), new Object[0]);
-        }
+			if (responseCode != 200) {
+				String errorMessage = con.getResponseMessage();
```

Comment:
I think the getResponseMessage() will just return "Not Found" for all 404 responses.. 

See: https://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#getResponseMessage()